include/mc_tasks/TrajectoryTaskGeneric.hpp

```diff
@@ -200,14 +200,19 @@ Eigen::VectorXd TrajectoryTaskGeneric<T>::dimWeight() const
 template<typename T>
 void TrajectoryTaskGeneric<T>::selectActiveJoints(
     const std::vector<std::string> & activeJointsName,
-    const std::map<std::string, std::vector<std::array<int, 2>>> & activeDofs)
+    const std::map<std::string, std::vector<std::array<int, 2>>> & activeDofs,
+    bool checkJoints)
 {
   if(inSolver_)
   {
     LOG_WARNING("selectActiveJoints(names) ignored: use selectActiveJoints(solver, names) for a task already added to "
                 "the solver");
     return;
   }
+  if(checkJoints)
+  {
+    ensureHasJoints(robots.robot(rIndex), activeJointsName, "[selectActiveJoints]");
```

Comment:
`name() + "::selectActiveJoints"` is a better prefix imho (applies to all other places where the prefix is passed)